src/graphql/schema/collection/createCollectionGqlType.ts

```diff
@@ -45,12 +46,17 @@ export default function createCollectionGqlType<TValue> (
       Array.from(type.fields).map(
         <TFieldValue>([fieldName, field]: [string, ObjectType.Field<TValue, TFieldValue>]) => {
           const { gqlType, intoGqlOutput } = getGqlOutputType(buildToken, field.type)
+          const formattedFieldName = formatName.field(fieldName)
           return {
-            key: formatName.field(fieldName),
+            key: formattedFieldName
             value: {
               description: field.description,
               type: gqlType,
-              resolve: (value: TValue): mixed => intoGqlOutput(field.getValue(value)),
+              sqlName: () => fieldName,
```

Comment:
I agree with the desire for separation (I've not bothered yet because I was concentrating on just getting *something* working, but I will be looking at it during refactoring); but I think a full two-stage parse of the entire AST is a bit ambitious at this stage - I'd rather get a step in the right direction out than wait for perfection that may never come. (Incidentally this is the approach JoinMonster takes: transforming the entire AST and using it to build the SQL queries)